Admin Actions Log: `pulsar-gpp-compiler` (#198)
pulsar-edit/package-backend: docs/reference/Admin_Actions.md

With that said this document will serve as the ongoing history of administrative actions that must be taken against the backend.

## 2023 - September 4

### pulsar-gpp-compiler

The maintainers of the community package `pulsar-gpp-compiler` during the course of maintaining their package, had accidentally completely unpublished the package, rather than unpublishing a specific version.

> These maintainers had wanted to retain the same package name during a republication, but the backend reserves all package names after unpublishing to help keep users safe from a potentially [Supply Chain Vulnerability](https://en.wikipedia.org/wiki/Supply_chain_attack).

Reviewer: ```suggestion
These maintainers had wanted to retain the same package name during a republication, but the backend reserves all package names after unpublishing to help keep users safe from a potential [Supply Chain Vulnerability](https://en.wikipedia.org/wiki/Supply_chain_attack).
```

At the request of this package's maintainers we investigated any way to allow the same name to be published, while keeping our users safe.

Since the nature of a Supply Chain Vulnerability relies on users having previously installed the potential package, and due to the young age of the package, we realized that if we can confirm with 100% certainty that absolutely 0 users had installed this package, nobody could be affected by a supply chain vulnerability, and it could be safe to the overall community to allow republication.

An important note about confirming the download count: Because when a package is unpublished, the download count is permanently deleted from the Pulsar Package Repository, we have to rely on the logs of the backend. These logs only go back 30 days, meaning that if a package was published more than 30 days ago, it would **never** be eligible for allowing republication of a reserved name.

Beyond that we had to confirm the following details:

> 1. The user requesting this service had sufficient permission to the effected repository.

Reviewer: ```suggestion
1. The user requesting this service had sufficient permission to the affected repository.
```

Reviewer: sorry one last one I found

Author: Thanks for taking a second look

> 2. No major, malicious changes had occurred in the codebase, which could signal a GitHub users account has been compromised.

Reviewer: ```suggestion
2. No major, malicious changes had occurred in the codebase, which could signal a GitHub user's account had been compromised.
```

3. That absolutely 0 downloads have ever occurred of this package.

To confirm the above:

> 1. The user was asked to create a branch of the effected repo named `allow-pulsar-un-reserving`

Reviewer: ```suggestion
1. The user was asked to create a branch of the affected repo named `allow-pulsar-un-reserving`
```
